### PR TITLE
Overhaul flashing docs once again

### PR DESCRIPTION
**Pull request by marcelstoer**
Addresses https://github.com/nodemcu/nodemcu-firmware/pull/1585#issuecomment-259273835.

### CODE REVIEW

File: docs/en/flash.md [hunk 1]
```diff
@@ -28,7 +28,9 @@ Run the following command to flash an *aggregated* binary as is produced for exa
 - `size` is given in bits. Specify `4m` for 512&nbsp;kByte and `32m` for 4&nbsp;MByte.  If unsure, try a smaller size;
```
Comment (pjsg) [inline docs/en/flash.md]: Modern versions of the esptool.py actually auto-detect the flash size, so it is probably best to leave out the flash size. This functionality was added about a month ago. We may want to note the flash size issues in some notes somewhere.

File: docs/en/flash.md [hunk 2]
```diff
@@ -41,6 +43,8 @@ Source: [https://github.com/nodemcu/nodemcu-flasher](https://github.com/nodemcu/
 
 Supported platforms: Windows
 
+Note that this tool was created by the initial developers of the NodeMCU firmware. It hasn't seen updates since September 2015 and is not maintained by the current NodeMCU *firmware* team. Be careful not to accidentally flash the very old default firmware the tool is shipped with.
```
Comment (pjsg) [inline docs/en/flash.md]: This tool is actively maintained by espressif (I followed the link to it higher up in the document): https://github.com/espressif/esptool

File: docs/en/flash.md [hunk 3]
```diff
@@ -58,33 +62,57 @@ Otherwise, if you built your own firmware from source code:
...
 
-* The NodeMCU flash filesystem may need to be reformatted, particularly if its address has changed because the new firmware has a much different size than the old firmware.  If it is not automatically formatted, then it should be valid and have the same contents as before the flash operation. You can still run [`file.format()`](modules/file.md#fileformat) to re-format your flash filesystem. You will know if you need to do this if your flash files exist but seem empty, or if data cannot be written to new files. However, this should be an exceptional case.
+* The NodeMCU flash file system may need to be reformatted, particularly if its address has changed because the new firmware is different in size from the old firmware.  If it is not automatically formatted then it should be valid and have the same contents as before the flash operation. You can still run [`file.format()`](modules/file.md#fileformat) manually to re-format your flash file system. You will know if you need to do this if your flash files exist but seem empty, or if data cannot be written to new files. However, this should be an exceptional case.
```
Comment (marcelstoer) [inline docs/en/flash.md]: Ok, I can certainly do that.

File: docs/en/flash.md (hunk 1 above)
Comment (marcelstoer) [inline docs/en/flash.md]: Oh nice, just found the [PR](https://github.com/espressif/esptool/pull/131), thanks. Don't you think it'd make sense to mention that only for [esptool.py >= 1.2](https://github.com/espressif/esptool/releases) you can skip the flash size parameter?

Comment (pjsg) [inline docs/en/flash.md]: I think that I would recommend leaving the parameter out. However, you should probably note that the esptool is under active development and that if it doesn't work, the first step should be to upgrade to the current version.

**Comment by marcelstoer** [comment]
Philip, I processed your feedback and pushed a new version.

**Comment by marcelstoer** [comment]
I take your thumbs-up as a go for merging.

**Comment by pjsg** [comment]
Yes!

**Pull request merged**